src/components/DetailsModal.tsx

```diff
@@ -40,9 +51,16 @@ function DetailsModal({
   );
   const auth = useAuth();
   const [details, setDetails] = useState<SubmissionDetails>({
-    upload_id: submission.upload_id,
-    data_stream_id: submission.metadata?.data_stream_id,
-    data_stream_route: submission.metadata?.data_stream_route,
+    status: submission.status,
+    lastService: "",
+    lastAction: "",
+    filename: submission.filename,
+    uploadId: submission.upload_id,
+    dexIngestTimestamp: submission.timestamp,
+    dataStreamId: submission.metadata?.data_stream_id,
+    dataStreamRoute: submission.metadata?.data_stream_route,
+    jurisdiction: submission.jurisdiction,
+    senderId: submission.sender_id,
```

Comment:
Can you just do `...submission`?

Reply:
Spread operators in javascript on objects are very sus. They only shallow copy, so many times lower level items don't actually get copied.

Reply:
Also all of the keys don't have the same names.